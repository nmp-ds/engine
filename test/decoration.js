import { setup } from "./_helpers.js";
import { describe, expect, test } from "vitest";

setup();

describe("text decoration", () => {
  test("supports underline and line-through", async (t) => {
    const classes = ['underline', 'line-through']

    const { css } = await t.uno.generate(classes);

    expect(css).toMatchSnapshot();
  });

  test("supports no-underline", async (t) => {
    const classes = ["no-underline"];

    const { css } = await t.uno.generate(classes);

    expect(css).toMatchSnapshot();
  });
})
