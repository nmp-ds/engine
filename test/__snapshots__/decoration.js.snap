// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`text decoration > supports no-underline 1`] = `
"/* layer: default */
.no-underline{text-decoration:none;}"
`;

exports[`text decoration > supports underline and line-through 1`] = `
"/* layer: default */
.line-through{text-decoration-line:line-through;}
.underline{text-decoration-line:underline;}"
`;
